src/main/java/ui/UI.java

```diff
@@ -174,6 +183,9 @@ private void showMainWindow(String repoId) {
         // Should only be called after panels have been initialized
         ensureSelectedPanelHasFocus();
         initialisePickers();
+        if (shouldCreateSampleBoard && (!TestController.isTestMode() || TestController.isTestStartupBoard())){
```

Comment:
Startup board testing is required only on app launch. Other tests go with the usual assumption of bypassing login and testing HubTurbo features assuming a repo has been created. Thus, we need a flag to test the startup board feature. Makes it neater. Yup, it's helpful to explain that. I have added a comment about it too.

`logic.openPrimaryRepository`/`logic.setDefaultRepo` change `logic.getStoredRepos().isEmpty()` so this needs to be tested before those two. The number of stored repos will be 0 the first time a user logs in (i.e.) if the store folder is empty. So the check is needed at that place.